content/partials/code-signing-personal-ios.md

```diff
@@ -1,25 +1,29 @@
 ---
 ---
-{{<markdown>}}
-#### Creating the App Store Connect API key
-Signing iOS applications requires [Apple Developer Program](https://developer.apple.com/programs/enroll/) membership.
-{{</markdown>}}
-{{< include "/partials/app-store-connect-api-key.md" >}}
 
-{{< include "/partials/code-signing-ios-obtain-certificate.md" >}}
+### Automatic vs Manual code signing
 
-{{< include "/partials/code-signing-ios-configure-environment-vars.md" >}}
-
-#### Automatic vs Manual code signing
+{{<notebox>}}
+Signing iOS applications requires [Apple Developer Program](https://developer.apple.com/programs/enroll/) membership. 
+{{</notebox>}}
 
 Signing iOS apps requires a `signing certificate` (App Store **development** or **distribution** certificate in `.p12` format) and a `provisioning profile`. In **manual code signing** you save these files as Codemagic `environment variables` and manually reference them in the appropriate build steps.
 
 In **Automatic code signing**, Codemagic takes care of Certificate and Provisioning profile management for you. Based on the `Certificate private key` that you provide, Codemagic will automatically fetch the correct certificate from the App Store or create a new one if necessary.
 
-#### Automatic code signing
+### Automatic code signing
+
+When automatic code signing is used, then most up-to-date signing files are obtained directly from Apple during the build time. This requires that Codemagic has access to your Apple Developer portal account, which is achieved by using App Store Connect API key.
 
-To code sign the app, add the following commands in the [`scripts`](../getting-started/yaml#scripts) section of the configuration file, after all the dependencies are installed, right before the build commands. 
+#### Creating the App Store Connect API key
+
+{{< include "/partials/app-store-connect-api-key.md" >}}
+
+{{< include "/partials/code-signing-ios-obtain-certificate.md" >}}
+
+{{< include "/partials/code-signing-ios-configure-environment-vars.md" >}}
```

Comment:
Including this here makes the whole automatic code signing part extremely difficult to follow for me, mainly due to sections being nested inside sections. It remains unclear why the Apple Developer Portal integration section is part of "Configuring environment variables". I believe the only reason for this is to cover adding the private key as an env var? And why is that even needed if we're using the integration to authenticate?

Reply:
Certificate private key has nothing to do with ASC/ADP authentication and is a standalone thing that is needed whenever working with certificates that are downloaded from Dev portal as for code signing one need both the secret part (certificate private key) and the certificate itself. Apple gives you only the latter while the former needs comes from user.
In case of manual signing certs are stored as PKCS#12 containers which have both certificate and private key within one single file.